packages/xrpl/src/Wallet/index.ts

```diff
@@ -166,19 +170,60 @@ class Wallet {
   public static fromSecret = Wallet.fromSeed
 
   /**
-   * Derives a wallet from a mnemonic.
+   * Derives a wallet from an entropy (array of random numbers).
+   *
+   * @param entropy - An array of random numbers to generate a seed used to derive a wallet.
+   * @param opts - (Optional) Options to derive a Wallet.
+   * @param opts.algorithm - The digital signature algorithm to generate an address for.
+   * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
+   * @returns A Wallet derived from an entropy.
+   */
+  public static fromEntropy(
+    entropy: Uint8Array | number[],
+    opts: { masterAddress?: string; algorithm?: ECDSA } = {},
+  ): Wallet {
+    const algorithm = opts.algorithm ?? DEFAULT_ALGORITHM
+    const options = {
+      entropy: Uint8Array.from(entropy),
+      algorithm,
+    }
+    const seed = generateSeed(options)
+    return Wallet.deriveWallet(seed, {
+      algorithm,
+      masterAddress: opts.masterAddress,
+    })
+  }
+
+  /**
+   * Derives a wallet from a bip39 or RFC1751 mnemonic (Defaults to bip39).
    *
    * @param mnemonic - A string consisting of words (whitespace delimited) used to derive a wallet.
    * @param opts - (Optional) Options to derive a Wallet.
-   * @param opts.derivationPath - The path to derive a keypair (publicKey/privateKey) used for mnemonic-to-seed conversion.
    * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
+   * @param opts.derivationPath - The path to derive a keypair (publicKey/privateKey). Only used for bip39 conversions.
+   * @param opts.useRFC1751 - If true, this interprets the mnemonic as a rippled RFC1751 mnemonic like `wallet_propose`
+   *                          generates in rippled.
+   * @param opts.algorithm - Only used if opts.useRFC1751Mnemonic is true. Allows the mnemonic to generate its
+   *                         secp256k1 seed, or its ed25519 seed.
```

Comment:
I don't love that there are 2 options for this - it feels like there should be one option with 3 possible values: bip39, RFC1751-with-secp, RFC1751-with-ed25519 (obviously with better naming conventions). Happy to discuss.

Reply:
I think regardless you end up with awkwardness (Since `derivationPath` is an optional param which is only relevant for bip39)

This also allows us to set the default to match `wallet_propose` which defaults to secp256k1 as opposed to our usual default of ed25519.